dotcom-rendering/src/web/components/ShowHideContainer.importable.tsx

```diff
@@ -0,0 +1,98 @@
+import { css } from '@emotion/react';
+import { from, space, neutral, textSans } from '@guardian/source-foundations';
+import { ButtonLink } from '@guardian/source-react-components';
+import { useEffect, useState } from 'react';
+import { storage } from '@guardian/libs';
+
+type Props = {
+	sectionId: string;
+	overrideContainerToggleColour?: string;
+};
+
+type ContainerStates = { [id: string]: string };
+
+const showHideButtonCss = (
+	overrideContainerToggleColour: string | undefined,
+) => {
+	return css`
+		color: ${neutral[46] || overrideContainerToggleColour};
+		${textSans.xsmall()};
+	`;
+};
+
+const showHideContainerCss = css`
+	margin-top: ${space[2]}px;
+	margin-right: 10px;
+	margin-bottom: ${space[2]}px;
+	position: relative;
+	align-items: bottom;
+
+	${from.wide} {
+		position: absolute;
+		top: 0;
+		right: 0;
+	}
+`;
+
+const getContainerStates = (): ContainerStates => {
+	const item = storage.local.get(`gu.prefs.container-states`);
```

Comment:
Hm yes... I hadn't thought of that... I don't think the overhead from reading from localstorage is all that much, but hydrating it will be... Maybe I can turn this component into something similar to the Liveness component for Liveblogs so we only load it in once and use it to hydrate all of the containers on the page.

Reply:
> Maybe I can turn this component into something similar to the Liveness component for Liveblogs so we only load it in once and use it to hydrate all of the containers on the page

💯 I think this is going to be the most performant option

I also think you're right that the read from LS won't necessarily be onerous but that hydration of multiple islands is another matter

Reply:
Should all be hydrated from a centralized source now!